src/config/config_api.c

```diff
@@ -104,3 +103,328 @@ err:		__wt_free(session, config_parser);
 
 	return (ret);
 }
+
+/*
+ * Translation table of the API methods supported by wiredtiger_config_validate
+ * and WT_CONNECTION.configure_method APIs.
+ */
+static const struct {
+	const char *method;
+	const char *config;
+} name_to_config_list[] = {
+	{ "WT_CONNECTION.async_new_op", "connection.async_new_op" },
+	{ "WT_CONNECTION.close", "connection.close" },
+	{ "WT_CONNECTION.load_extension", "connection.load_extension" },
+	{ "WT_CONNECTION.open_session", "connection.open_session" },
+	{ "WT_CONNECTION.reconfigure", "connection.reconfigure" },
+	{ "WT_CURSOR.close", "cursor.close" },
+	{ "WT_CURSOR.reconfigure", "cursor.reconfigure" },
+	{ "WT_SESSION.begin_transaction", "session.begin_transaction" },
+	{ "WT_SESSION.checkpoint", "session.checkpoint" },
+	{ "WT_SESSION.close", "session.close" },
+	{ "WT_SESSION.commit_transaction", "session.commit_transaction" },
+	{ "WT_SESSION.compact", "session.compact" },
+	{ "WT_SESSION.create", "session.create" },
+	{ "WT_SESSION.drop", "session.drop" },
+	{ "WT_SESSION.open_cursor", "session.open_cursor" },
+	{ "WT_SESSION.reconfigure", "session.reconfigure" },
+	{ "WT_SESSION.rename", "session.rename" },
+	{ "WT_SESSION.rollback_transaction", "session.rollback_transaction" },
+	{ "WT_SESSION.salvage", "session.salvage" },
+	{ "WT_SESSION.strerror", "session.strerror" },
+	{ "WT_SESSION.truncate", "session.truncate" },
+	{ "WT_SESSION.upgrade", "session.upgrade" },
+	{ "WT_SESSION.verify", "session.verify" },
+	{ "wiredtiger_open", "wiredtiger_open" },
+};
```

Comment:
How much would break if we pushed this renaming all the way into `api_data.py` (and the `API_CALL` macros), so we never have <lower-case-handle>.<method>?